### PR TITLE
[patch] AWS support: SLS

### PR DESCRIPTION
**Pull request by prasanthgelli1**
This change seperates the installation of SLS (`sls_install` role) from the generation of the SLSCfg used by MAS (`sls_gencfg` role).

The sls_install role will generate the varaibles used by the sls_gencfg role, so that the two roles can be easily chained together when needed, but this also allows the gencfg role to be called in isolation to easily generate a SLSCfg  definition for an existing SLS instance.

### CODE REVIEW

File: ibm/mas_devops/playbooks/dependencies/install-sls.yml [hunk 1]
```diff
@@ -35,15 +35,16 @@
...
+    sls_cfg_file: "{{ mas_config_dir }}/sls.yml"
+
+    # configuration file for
+    mongodb_cfg_file: "{{mas_config_dir}}/mongo-{{ lookup('env', 'MONGODB_NAMESPACE') }}.yml"
 
     # Bootstrap is used to initialize SLS, provide if you have an existing lic file and licenseId
     bootstrap:
-      license_id: "{{ lookup('env', 'SLS_LICENSE_ID') }}"
-      registration_key: "{{ lookup('env', 'SLS_REGISTRATION_KEY') }}"
+      license_id: "{{ lookup('env', 'SLS_LICENSE_ID') | default('', true) }}"
```
Comment (durera) [inline ibm/mas_devops/playbooks/dependencies/install-sls.yml:46]: Don't use `| default('', true)` follow an env lookup ... it's redunant, the default value if the env var doesn't exist is already an empty string.

File: ibm/mas_devops/roles/sls_cfg/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+sls_cfg
```
Comment (durera) [inline ibm/mas_devops/roles/sls_cfg/README.md:1]: What is this new role doing?  What are the reasons for breaking this out into it's own role, please provide information in the PR / in this thread.

File: ibm/mas_devops/roles/sls_cfg/templates/slscfg.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
```
Comment (durera) [inline ibm/mas_devops/roles/sls_cfg/templates/slscfg.yml.j2:1]: None of the variables referenced in this template are configured inside this role.

File: ibm/mas_devops/roles/sls_install/templates/credential-request.yml.j2 [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    kind: AWSProviderSpec
+    statementEntries:
+    - effect: Allow
+      action:
+      - license-manager:CheckoutLicense
+      - license-manager:GetLicense
+      - license-manager:CheckInLicense
+      - license-manager:ExtendLicenseConsumption
+      - license-manager:ListReceivedLicenses
+      resource: "*"
```
Comment (durera) [inline ibm/mas_devops/roles/sls_install/templates/credential-request.yml.j2:22]: missing a newline in the file, atm linting ignored j2 files, but it won't forever, please add newline to end of file

File: ibm/mas_devops/roles/sls_install/tasks/main.yml [hunk 5]
```diff
@@ -161,12 +162,24 @@
...
 - include_tasks: tasks/slscfg.yml
   when:
     - mas_instance_id is defined
-    - mas_instance_id != ""
+    - mas_instance_id|length > 0
     - mas_config_dir is defined
-    - mas_config_dir != ""
+    - mas_config_dir|length > 0
+
+# 10. Create Cloud CredentialsRequest in SLS namespace
```
Comment (durera) [inline ibm/mas_devops/roles/sls_install/tasks/main.yml:180]: What is this for, what does it do?  Provide comments in this task and update the role README with details.

File: ibm/mas_devops/roles/sls_install/tasks/main.yml [hunk 6]
```diff
@@ -161,12 +162,24 @@
 # -----------------------------------------------------------------------------
 - include_tasks: "tasks/sls-verify.yml"
 
+# Adding some delay to get SLS LicenseService ready
```
Comment (durera) [inline ibm/mas_devops/roles/sls_install/tasks/main.yml:165]: Do not add any fixed delays like this.  The code works today without any, there is no reason to add a fixed 5 minute delay,we already have code that waits until SLS is ready by polling the state of the CR.  

The role is idempotent, if I run the role after successfully installing it should execute and complete super fast because SLS is already installed.  With this change we just added 5 minutes even when the role needs to do nothing.

File: ibm/mas_devops/roles/sls_install/tasks/main.yml [hunk 7]
```diff
@@ -3,12 +3,12 @@
 # 1. Check for missing properties that do not have defaults
 # -----------------------------------------------------------------------------
 - name: "Fail if sls_entitlement_username has not been provided"
-  when: sls_entitlement_username is not defined or sls_entitlement_username == ""
+  when: sls_entitlement_username is not defined or sls_entitlement_username|length == 0
```
Comment (durera) [inline ibm/mas_devops/roles/sls_install/tasks/main.yml:6]: What's the reason behind this change?  Also, while updating these checks please convert them to asserts while doing so (see #124)

File: ibm/mas_devops/roles/sls_install/defaults/main.yml [hunk 8]
```diff
@@ -7,6 +7,8 @@ sls_channel: 3.x
 # By default we will install into this namespace
 sls_namespace: ibm-sls
 
+sls_cfg_file: /tmp/slscfg-local-system.yaml
```
Comment (durera) [inline ibm/mas_devops/roles/sls_install/defaults/main.yml:10]: Why has this been added/for what purpose?

We use a generated value for sls_cfg_file of `"{{ mas_config_dir }}/sls-{{ sls_instance_name | lower }}-{{ sls_namespace }}.yml"`

**Review by durera — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/sls_cfg/README.md (hunk 2 above)
Comment (prasanthgelli1) [reply ibm/mas_devops/roles/sls_cfg/README.md:1]: This role is used to generate a configuration file incase of an existing sls.

File: ibm/mas_devops/roles/sls_cfg/templates/slscfg.yml.j2 (hunk 3 above)
Comment (prasanthgelli1) [reply ibm/mas_devops/roles/sls_cfg/templates/slscfg.yml.j2:1]: We are configuring  them in cfg-sls playbook, and we are validating them in task/main.yml.

File: ibm/mas_devops/roles/sls_install/tasks/main.yml (hunk 5 above)
Comment (prasanthgelli1) [reply ibm/mas_devops/roles/sls_install/tasks/main.yml:180]: We are validating the length of mas_config_dir over here.

File: ibm/mas_devops/playbooks/dependencies/install-sls.yml (hunk 1 above)
Comment (prasanthgelli1) [reply ibm/mas_devops/playbooks/dependencies/install-sls.yml:46]: Sure, Removing default statements.

File: ibm/mas_devops/roles/sls_install/tasks/main.yml (hunk 6 above)
Comment (prasanthgelli1) [reply ibm/mas_devops/roles/sls_install/tasks/main.yml:165]: Sure. Reverting back the change.

File: ibm/mas_devops/roles/sls_install/defaults/main.yml (hunk 8 above)
Comment (prasanthgelli1) [reply ibm/mas_devops/roles/sls_install/defaults/main.yml:10]: We wanted to use this tmp file for configuration. But i see that it is getting generated. Removing it.

**Comment by prasanthgelli1** [comment]
@durera  Made the changes you asked for. Can you please verify again.

**Review by prasanthgelli1 — COMMENTED** [review]
[minor] Merge changes MAS ON AWS

**Comment by prasanthgelli1** [comment]
@durera All the checks have passed and the build has been successful please verify.

File: ibm/mas_devops/roles/sls_cfg/README.md (hunk 2 above)
Comment (durera) [reply ibm/mas_devops/roles/sls_cfg/README.md:1]: DIscussed in Slack, this is a different usecase, we will move all slscfg mgmt into a seperate role so that it is no longer duplicated in the sls_install role

**Review by durera — APPROVED** [review]
